site/en/blog/prerender-pages/index.md

```diff
@@ -0,0 +1,271 @@
+
+Below is a simple example of adding a `prerender` speculation rule:
+
+```js
+if (HTMLScriptElement.supports &&
+    HTMLScriptElement.supports('speculationrules')) {
+  const specScript = document.createElement('script');
+  specScript.type = 'speculationrules';
+  specRules = {
+    'prerender': [
+      {
+        source: 'list',
+        urls: ['/next.html'],
+      },
+    ],
+  };
+  specScript.textContent = JSON.stringify(specRules);
+  console.log('added speculation rules to: next.html');
+  document.body.append(specScript);
+}
+```
+
+You can view a demo of Speculation Rules API prerendering on [this demo page](https://prerender-demos.glitch.me/).
+
+## Detecting and disabling prerendering
+
+Prerender is usually a positive experience for users as it allows fast page rendering—often instant. This benefits both the user, and the site owner, since prerendered pages allow a better user experience that may be difficult to achieve otherwise.
+
+However, there may be instances when you do not wish prerendering of pages to happen, for example when pages change state—either based on the initial request, or based on JavaScript executing on the page.
+
+### Enabling or disabling prerender in Chrome
+
+Prerender is only enabled for those Chrome users with the "Preload pages for faster browsing and searching" setting in `chrome://settings/cookies/`. Additionally, prerender is also disabled on low-memory devices, or if the operating system is in data-saver or battery-saver modes.
+
+### Detecting prerender server-side
+
+Prerendered pages will be sent with the following HTTP header:
+
+```http
+Sec-Purpose: prefetch;prerender
+```
+
+Servers can respond based on this header, to log prerender requests, return different content, or prevent the prerender from happening. If a non-success response code is returned (that is, not a 200 or a 304), then the page will not be prerendered. So this is one way to prevent pages you absolutely do not want prerendered, from prerendering.
+
+### Detecting prerender in JavaScript
+
+The `document.prerendering` API will return `true` while the page is prerendering. This can be used by pages to prevent—or delay—certain activities during the prerender until the page is actually activated.
+
+When the page is activated by the user viewing the page, the `prerenderingchange` event will be dispatched on the `document`, which can then be used to enable activities that previously would be started by default on page load but which you wish to delay until the page is actually viewed by the user.
+
+Once a prerendered document is activated, `PerformanceNavigationTiming`'s `activationStart` will also be set to a non-zero time representing the time between when the prerender was started and the document was actually activated.
```

Comment:
I think this section needs a code snippet that combines this and the previous paragraph into a single function. E.g.:

```js
function wasPagePrerendered() {
  return document.prerendering || self.performance?.getEntriesByType?.call(
      performance, 'navigation')[0]?.activationStart > 0;
}
```

Alternatively, if you want a function that _only_ returns true if the page _was_ prerendered (and not is currently prerendering), then you could make this an async function that awaits the `prerenderingchange` event if the page is currently prerendering.

Reply:
Added that but changed it to `pagePrerendered` as didn't like the "was" when it might still be prerendering.